pkg/monitor/monitor.go

```diff
@@ -396,3 +458,25 @@ func (m *Monitor) cleanupVolumeAttachmentsByNode(node *volume.NodeInfo) error {
 
 	return nil
 }
+
+func (m *Monitor) getVolumeDriverNodesToK8sNodeMap(driverNodes []*volume.NodeInfo) map[string]*v1.Node {
+	nodeMap := make(map[string]*v1.Node)
+
+	k8sNodes, err := core.Instance().GetNodes()
+	if err != nil {
+		log.Errorf("Error getting nodes in monitor: %v", err)
+		return nodeMap
```

Comment:
It's not an infinite loop though. For every cycle of monitoring, it will go over the nodes and only if the node status is offline, then we are trying to put that in the map. In one cycle of drivermonitor (default value of 2mins) , the k8s call happens one time to create the map (for successful case) but if k8s itself does not response, then that call can happen for offline number of nodes. 

Inside getVolumeDriverNodesToK8sNodeMap , there is no API calls to PX.

Reply:
Ok. I see a sleep after the inner for loop. It should be ok then.